File: source/kCura.Relativity.Export/Types/CoalescedTextViewField.vb
Imports RelativityExportConstants = Relativity.Export.Constants

Namespace kCura.Relativity.Export.Types
	Public Class CoalescedTextViewField
		Inherits ViewFieldInfo
		Public Sub New(ByVal vfi As ViewFieldInfo, ByVal useCurrentFieldName As Boolean)
			MyBase.New(vfi)
			_avfColumnName = RelativityExportConstants.TEXT_PRECEDENCE_AWARE_AVF_COLUMN_NAME
			Dim nameToUse As String = RelativityExportConstants.TEXT_PRECEDENCE_AWARE_AVF_COLUMN_NAME
			If useCurrentFieldName Then nameToUse = vfi.DisplayName
			_avfHeaderName = nameToUse
			_displayName = nameToUse
		End Sub
	End Class
End Namespace

File: source/kCura.Relativity.Export.ConvertedToC#/Types/CoalescedTextViewField.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Data;
using System.Linq;
using RelativityExportConstants = Relativity.Export.Constants;

namespace kCura.Relativity.Export.Types
{
	public class CoalescedTextViewField : ViewFieldInfo
	{
		public CoalescedTextViewField(ViewFieldInfo vfi, bool useCurrentFieldName) : base(vfi)
		{
			_avfColumnName = RelativityExportConstants.TEXT_PRECEDENCE_AWARE_AVF_COLUMN_NAME;
			string nameToUse = RelativityExportConstants.TEXT_PRECEDENCE_AWARE_AVF_COLUMN_NAME;
			if (useCurrentFieldName)
				nameToUse = vfi.DisplayName;
			_avfHeaderName = nameToUse;
			_displayName = nameToUse;
		}
	}
}
